### ISSUE TITLE
Segunda entrega

### ISSUE THREAD

**Issue by DamRCorba** [martinrosolino/TPinfo2#1]
Aprobada, podes seguir con el codigo, recorda que cada estado se reemplaza por una funcion de estado que tiene que devolver un estado.

**Comment by martinrosolino** [comment]
Buenas profe, quería saber si pudo ver los tres archivos con código para saber si tengo algún error. Yo probé todas las situaciones posibles y funcionó bien pero capaz se me está pasando algo. Saludos.

**Comment by DamRCorba** [comment]
Para la entrega de hoy esta bien. No deberias usar variables globales.

**Comment by DamRCorba** [comment]
Martín, no cumple con la premisa de que cada estado debe ser representado por una función de estado que devuelve un estado. La implementación que hiciste de la función inicio desvía al código de lo que realmente debería hacer. Cada estado debería ser el encargado de decidir cual es el estado siguiente a ejecutar, Y esto es porque en una maquina de estado podrías tener un mismo estado al cual se llegue desde diferentes estados y este debería ser capaz de encaminar el circuito del código. En tu implementación esta característica se pierde, si bien la podrías reemplazar pasando como parámetro el estado anterior a tu función inicio, esto no resulta óptimo. Deberías hacer que cada función de estado devuelva un estado.

**Comment by DamRCorba** [comment]
Hola Martin, aun no cierra el codigo. Recorda que los estados que devolves tienen que ser con etiquetas y no con numeros. Ademas, ya que tenes cuestiones temporales deberias analizar de utilizar la libreria Time.h

**Comment by martinrosolino** [comment]
Buenas profe, ya corregí el tema de las etiquetas y parte de time.h. Ya pude obtener la hora del sistema para que chequee el horario y analice el estado del puente en ese momento. El problema me surgió cuando quise implementar el control manual. Hay alguna forma de, por ejemplo, hacer un scanf y si no se ingresa nada por 60 segundos que deje de leer o algo así?
Muchas gracias.

**Comment by DamRCorba** [comment]
No, pero podrias usar la funcion kbhit que chequea si se presiona una tecla del teclado para leerla despues con el scanf.

    El martes, 15 de septiembre de 2020 11:08:41 ART, Martín Rosolino <notifications@github.com> escribió:  
 
 


Buenas profe, ya corregí el tema de las etiquetas y parte de time.h. Ya pude obtener la hora del sistema para que chequee el horario y analice el estado del puente en ese momento. El problema me surgió cuando quise implementar el control manual. Hay alguna forma de, por ejemplo, hacer un scanf y si no se ingresa nada por 60 segundos que deje de leer o algo así?
Muchas gracias.

**Comment by martinrosolino** [comment]
Perfecto, acabo de implementar esa función para que detecte si se presionó o no una tecla para alternar el estado. Ya pushee los archivos nuevos.

**Comment by DamRCorba** [comment]
Perfecto Martin. Ya lo revise, TP aprobado.
